src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/Order2DeliveryForm.tsx

```diff
@@ -34,12 +52,89 @@ export const Order2DeliveryForm: React.FC<Order2DeliveryFormProps> = () => {
     phoneNumber: "",
     phoneNumberCountryCode: "",
   }
-  const onSubmit = (
-    values: FormikValues,
-    formikHelpers: FormikHelpers<FormikValues>,
-  ) => {
-    checkoutTracking.clickedOrderProgression(ContextModule.ordersFulfillment)
-  }
+  const onSubmit = useCallback(
+    async (
+      values: AddressFormValues,
+      formikHelpers: FormikHelpers<FormikValues>,
+    ) => {
+      try {
+        checkoutTracking.clickedOrderProgression(
+          ContextModule.ordersFulfillment,
+        )
+        const updateShippingAddressResult =
+          await updateShippingAddressMutation.submitMutation({
+            variables: {
+              input: {
+                id: orderData.internalID,
+                buyerPhoneNumber: values.address.phoneNumber,
+                shippingAddressLine1: values.address.addressLine1,
+                shippingAddressLine2: values.address.addressLine2,
+                shippingCity: values.address.city,
+                shippingRegion: values.address.region,
+                shippingPostalCode: values.address.postalCode,
+                shippingCountry: values.address.country,
+                shippingName: values.address.name,
+              },
+            },
+          })
+        const updateShippingAddressOrder = validateAndExtractOrderResponse(
+          updateShippingAddressResult.updateOrderShippingAddress?.orderOrError,
+        ).order
+
+        // By this point the order should have a proper shipping option
+        // DOMESTIC_FLAT or INTERNATIONAL_FLAT: flat shipping, save it right away
+        // SHIPPING_TBD: artsy shipping, user will be presented
+        // with options on the next step
+        const fulfillmentOptions = updateShippingAddressOrder.fulfillmentOptions
+
+        const flatShippingOption =
+          fulfillmentOptions.find(
+            option =>
+              option.type === "DOMESTIC_FLAT" ||
+              option.type === "INTERNATIONAL_FLAT",
+          ) ?? null
+
+        const isFlatShipping = !!flatShippingOption
+
+        if (!isFlatShipping) {
+          const artsyShippingOption = fulfillmentOptions.find(
+            option => option.type === "SHIPPING_TBD",
+          )
+          logger.warn("hopefully artsy shipping...", !!artsyShippingOption)
+          return
+        }
+
+        const setFulfillmentOptionResult =
+          await setFulfillmentOptionMutation.submitMutation({
```

Comment:
Do I understand it correctly - if artwork has flat shipping and the address is set - this step will automatically select this flat shipping and just move on? If that is the case - maybe not worth additional client call and we could/should do it all in 'set address mutation' in exchange? 

Though looking at the figma (https://www.figma.com/design/kaBdUapJIWs9YL9ATlPDzw/Checkout-Redesign?node-id=4162-49577&m=dev) seems like we do not want to select this 'flat shipping' automatically - we still want collectors to click 'Continue to Payment'?

Reply:
You are right. Updated to use the new step, now built out for this version.